apps/src/templates/teacherDashboard/RosterDialog.jsx

```diff
@@ -123,13 +143,18 @@ export default class RosterDialog extends React.Component {
           {locale.selectGoogleClassroom()}
         </h2>
         <div style={styles.content}>
-          {this.props.classrooms ?
-            <ClassroomList
-              classrooms={this.props.classrooms}
-              onSelect={this.onClassroomSelected}
-              selectedId={this.state.selectedId}
+          {this.props.loadError ?
+            <LoadError
+              error={this.props.loadError}
+              studioUrl={this.props.studioUrl}
             /> :
-            locale.loading()
+            this.props.classrooms ?
+              <ClassroomList
+                classrooms={this.props.classrooms}
+                onSelect={this.onClassroomSelected}
+                selectedId={this.state.selectedId}
+              /> :
+              locale.loading()
```

Comment:
I tend to prefer avoiding ternaries in JSX as I find it them difficult to read/follow (nested ternaries even more so).